modules/kinesis/src/main/scala/com.snowplowanalytics.snowplow.enrich.stream/KinesisEnrich.scala

```diff
@@ -168,26 +170,39 @@ object KinesisEnrich extends Enrich {
    * @return The JSON stored in DynamoDB
    */
   private def lookupDynamoDBResolver(
-    provider: AWSCredentialsProvider,
+    provider: AwsCredentialsProvider,
+    endpoint: String,
     region: String,
     table: String,
     key: String
   ): Either[String, String] = {
-    val dynamoDBClient = AmazonDynamoDBClientBuilder
-      .standard()
-      .withCredentials(provider)
-      .withEndpointConfiguration(new EndpointConfiguration(getDynamodbEndpoint(region), region))
+
+    val dynamoDBClient = DynamoDbAsyncClient
+      .builder()
+      .credentialsProvider(provider)
+      .endpointOverride(new URI(endpoint))
+      .region(Region.of(region))
       .build()
-    val dynamoDB = new DynamoDB(dynamoDBClient)
     for {
-      // getTable doesn't involve any IO apparently so it's safe to chain
-      item <- Option(dynamoDB.getTable(table).getItem("id", key))
-                .fold(s"Key $key doesn't exist in DynamoDB table $table".asLeft[Item])(_.asRight[String])
-      json <- Option(item.getString("json"))
-                .fold(s"""Field "json" not found at key $key in DynamoDB table $table""".asLeft[String])(
-                  _.asRight[String]
-                )
+      item <- Option(
+                dynamoDBClient
+                  .getItem(
+                    GetItemRequest
+                      .builder()
+                      .tableName(table)
+                      .key(Map[String, AttributeValue](("key", AttributeValue.builder().s(key).build())).asJava)
+                      .build()
+                  )
+                  .get(10, TimeUnit.SECONDS)
+                  .item()
+              ).fold(s"Key $key doesn't exist in DynamoDB table $table".asLeft[Map[String, AttributeValue]])(
+                _.asScala.toMap.asRight[String]
+              )
+      json <- item
+                .get("json")
+                .fold(s"""Field "json" not found at key $key in DynamoDB table $table""".asLeft[String])(_.s().asRight[String])
```

Comment:
I really liked `fold`. a good lesson for us to use to validate options in a fancy inline way.